crates/sc-consensus-subspace/src/verification.rs

```diff
@@ -199,15 +193,12 @@ fn check_piece<B: BlockT>(
 }
 
 /// Returns true if `solution.tag` is within the solution range.
-fn is_within_solution_range(
-    solution: &Solution,
-    global_challenge: [u8; 8],
-    solution_range: u64,
-) -> bool {
-    let farmer_public_key_hash = crypto::sha256_hash(&solution.public_key);
-    let local_challenge = derive_local_challenge(global_challenge, farmer_public_key_hash);
-
-    let target = u64::from_be_bytes(local_challenge);
+fn is_within_solution_range(solution: &Solution, solution_range: u64) -> bool {
+    let target = u64::from_be_bytes(
+        solution.local_challenge[..8]
+            .try_into()
+            .expect("Signature is always bigger than tag; qed"),
```

Comment:
Can we avoid/abstract away the magic number `8` here by implementing a helper method for `Signature` like `as_solution_target()` or something?

Reply:
Good catch, I had it with named constant in the other place, fixed. It is a general signature type, so having `.as_solution_target()` on it would be weird. Probably a separate data type would make sense here.

Reply:
A constant would be good as well.